src/java/org/apache/cassandra/config/Config.java

```diff
@@ -216,35 +248,39 @@
      * Default is the same as the native protocol frame limit: 256Mb.
      * See AbstractType for how it is used.
      */
-    public int max_value_size_in_mb = 256;
+    @Replaces(oldName = "max_value_size_in_mb", converter = Converter.MegabytesDataStorageConverter.class, deprecated = true)
+    public DataStorage max_value_size = new DataStorage("256MB");
 
     public boolean snapshot_before_compaction = false;
     public boolean auto_snapshot = true;
     public volatile long snapshot_links_per_second = 0;
 
     /* if the size of columns or super-columns are more than this, indexing will kick in */
-    public int column_index_size_in_kb = 64;
-    public volatile int column_index_cache_size_in_kb = 2;
-    public volatile int batch_size_warn_threshold_in_kb = 5;
-    public volatile int batch_size_fail_threshold_in_kb = 50;
+    @Replaces(oldName = "column_index_size_in_kb", converter = Converter.KilobytesDataStorageConverter.class, deprecated = true)
+    public volatile DataStorage column_index_size = new DataStorage("64KB");
+    @Replaces(oldName = "column_index_cache_size_in_kb", converter = Converter.KilobytesDataStorageConverter.class, deprecated = true)
+    public DataStorage column_index_cache_size = new DataStorage("2KB");
+    @Replaces(oldName = "batch_size_warn_threshold_in_kb", converter = Converter.KilobytesDataStorageConverter.class, deprecated = true)
+    public DataStorage batch_size_warn_threshold = new DataStorage("5KB");
+    @Replaces(oldName = "batch_size_fail_threshold_in_kb", converter = Converter.KilobytesDataStorageConverter.class, deprecated = true)
+    public volatile DataStorage batch_size_fail_threshold = new DataStorage("50KB");
     public Integer unlogged_batch_across_partitions_warn_threshold = 10;
     public volatile Integer concurrent_compactors;
-    public volatile int compaction_throughput_mb_per_sec = 16;
-    public volatile int compaction_large_partition_warning_threshold_mb = 100;
-    public int min_free_space_per_drive_in_mb = 50;
+    @Replaces(oldName = "compaction_throughput_mb_per_sec", converter = Converter.MegabitsPerSecondBitRateConverter.class, deprecated = true)
```

Comment:
Even in the old naming scheme, I guess this would be `MegabytesPerSecondBitRateConverter`, although sticking w/ `16MiB/s` means you could also use `MebibytesPerSecondBitRateConverter`, given that's what was previously meant by "mb_per_sec".

Reply:
I think the best will be to be made explicit `MebibytesPerSecondBitRateConverter` and use mebi, etc everywhere for clarity and consistency. I will revise them at once. To confirm again  - I am working on the review feedback and pushing locally but I haven't pushed new commits here not to interrupt your pass. But I acknowledge the points and involve on the discussions to be sure we are on the same page.